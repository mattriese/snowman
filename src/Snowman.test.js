import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Snowman from './Snowman';

it('renders without crashing', function () {
	render(<Snowman />);
});

it('game ends if numWrong = 7', function () {
	const { debug, container } = render(<Snowman words={["apple"]} />);

	const buttons = document.querySelectorAll('button');

	const b = container.querySelector('button[value="b"]');
	fireEvent.click(b);
	const c = container.querySelector('button[value="c"]');
	fireEvent.click(c);
	const d = container.querySelector('button[value="d"]');
	fireEvent.click(d);
	const f = container.querySelector('button[value="f"]');
	fireEvent.click(f);
	const g = container.querySelector('button[value="g"]');
	fireEvent.click(g);
	const h = container.querySelector('button[value="h"]');
	fireEvent.click(h);
	const i = container.querySelector('button[value="i"]');
	fireEvent.click(i);

	debug(container);
	expect(buttons).not.toBeVisible();
	expect("You lose!").toBeInTheDocument();
	expect("apple").toBeInTheDocument();
})


it('losing game matches snapshot', function () {
	const { container } = render(<Snowman />);

	const b = container.querySelector('button[value="b"]');
	fireEvent.click(b);
	const c = container.querySelector('button[value="c"]');
	fireEvent.click(c);
	const d = container.querySelector('button[value="d"]');
	fireEvent.click(d);
	const f = container.querySelector('button[value="f"]');
	fireEvent.click(f);
	const g = container.querySelector('button[value="g"]');
	fireEvent.click(g);
	const h = container.querySelector('button[value="h"]');
	fireEvent.click(h);
	const i = container.querySelector('button[value="i"]');
	fireEvent.click(i);
	expect(container).toMatchSnapshot();
});


// it('works when you click on the right arrow', function () {
//   const { container } = render(<Snowman />);
