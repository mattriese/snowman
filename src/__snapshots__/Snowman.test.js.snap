// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`losing game matches snapshot 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="7"
    />
    <p>
      Number wrong: 
      7
    </p>
    <p
      class="Snowman-word"
    >
      _
      _
      _
      _
      _
    </p>
    <p>
      <button
        value="a"
      >
        a
      </button>
      <button
        disabled=""
        value="b"
      >
        b
      </button>
      <button
        disabled=""
        value="c"
      >
        c
      </button>
      <button
        disabled=""
        value="d"
      >
        d
      </button>
      <button
        value="e"
      >
        e
      </button>
      <button
        disabled=""
        value="f"
      >
        f
      </button>
      <button
        disabled=""
        value="g"
      >
        g
      </button>
      <button
        disabled=""
        value="h"
      >
        h
      </button>
      <button
        disabled=""
        value="i"
      >
        i
      </button>
      <button
        value="j"
      >
        j
      </button>
      <button
        value="k"
      >
        k
      </button>
      <button
        value="l"
      >
        l
      </button>
      <button
        value="m"
      >
        m
      </button>
      <button
        value="n"
      >
        n
      </button>
      <button
        value="o"
      >
        o
      </button>
      <button
        value="p"
      >
        p
      </button>
      <button
        value="q"
      >
        q
      </button>
      <button
        value="r"
      >
        r
      </button>
      <button
        value="s"
      >
        s
      </button>
      <button
        value="t"
      >
        t
      </button>
      <button
        value="u"
      >
        u
      </button>
      <button
        value="v"
      >
        v
      </button>
      <button
        value="w"
      >
        w
      </button>
      <button
        value="x"
      >
        x
      </button>
      <button
        value="y"
      >
        y
      </button>
      <button
        value="z"
      >
        z
      </button>
    </p>
  </div>
</div>
`;
